### ISSUE TITLE
Java Heap Space Out of Memory Error

### ISSUE THREAD

**Issue by davidkcoblentz** [IBM/zopeneditor-about#222]
Issue Type: <b>Bug</b>

I was editing COBOL source, I accidentally held down ENTER long enough to insert 10 or so blank lines, then hit CTRL+Z many times to reverse the 10 blank lines but instead got this error.

[Error - 7:27:15 PM] Request textDocument/documentLink failed.
  Message: Internal error.
  Code: -32603 
java.util.concurrent.CompletionException: java.lang.OutOfMemoryError: Java heap space
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:645)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:479)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.OutOfMemoryError: Java heap space

Extension version: 1.4.1
VS Code version: Code 1.64.2 (f80445acd5a3dadef24aa209168452a3d97cc326, 2022-02-09T22:02:28.252Z)
OS version: Windows_NT x64 10.0.17763
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6600U CPU @ 2.60GHz (4 x 2808)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.88GB (6.47GB free)|
|Process Argv|--crash-reporter-id 9f20c08b-1af9-440c-b8ce-de38eef3775d|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30425749
pythontb:30283811
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30438340
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30438360
pythonvs932:30410667
wslgetstarted:30433507
vsclayoutctrc:30437038
vsrem710cf:30416617
dsvsc009:30440023
vsbas813:30436447
vscscmwlcmt:30438805
helixcf:30438807

```

</details>

**Comment by schoko11** [comment]
Also reported here...(seems similar)
https://github.com/IBM/zopeneditor-about/issues/208

**Comment by davidkcoblentz** [comment]
Oh, sorry, I should have done a search first.  I was adding it to Github from VSCode while coding.

Thank you,
***@***.***

**Comment by phaumer** [comment]
Thanks. Related to #208 we did find a cause for unnecessary cpu and memory usage related to include file. We will have a look at this one as well.

**Comment by phaumer** [comment]
We believe this has been fix in Z Open Editor 2.0.0. Can you try again.
